71 tab enhancements

ashberg opened the pull request:
Fixes #71 and possibly #26.

ashberg committed:
Added styling to the tabs for when the width of the screen is less than 481px.

ashberg committed:
Working on resizing the 'tab's as the width of the 'tabs' component changes.

ashberg committed:
Added a tab theme for tabs that contain a tab header that contains a dropdown menu.

ashberg committed:
Cleaned up the css in tabs_demo.vue.

ashberg committed:
Working on resolving global coverage in the test_tabs.ts file.

ashberg committed:
Made the resize event handler for the window into a private function like in the context_menu component.

ashberg committed:
Moved 'Possible to manually select newly added tab' into its own describe function to evade the 'wrapper.exists' error caused by the WrapperComponent class.

james-perretta commented on src/components/tabs/tabs.vue:
```diff
@@ -108,6 +133,7 @@ export default class Tabs extends Vue {
         }
 
         element_data.class.push(
+          'tab-style',
```
Does this get used?

james-perretta commented on src/components/tabs/tabs.vue:
```diff
@@ -145,7 +171,7 @@ export default class Tabs extends Vue {
     );
 
-    return create_element('div', {}, header_elts);
+    return create_element('div', { ref: 'headers_container' }, header_elts);
```
Does this ref get used?

james-perretta commented on src/demos/color_palette_demo.vue:
```diff
@@ -13,6 +18,11 @@
       <p class="color-square pebble-medium"></p>
     </div>
 
+    <div class="color">
```
Where does dim fit in the pebble colors?

james-perretta commented on src/styles/colors.scss:
```diff
@@ -1,9 +1,13 @@
 
 $pebble-medium: hsl(210, 12%, 93%);
 
+$pebble-dim: hsl(210, 11%, 90%);
```
Can we come up with another name?

james-perretta committed:
Minor refactoring.
Updated tslint settings to ignore naming convention for vue lifecycle hooks.
In tabs test and multi file viewer tests, restored matchMedia to original version in afterEach.
Replaced pebble-dim and pebble-lighter colors with pebble-medium and pebble-light.
